### ISSUE TITLE
Implement `InputType` for `Box<str>`.

### ISSUE THREAD

**Issue by BratSinot** [async-graphql/async-graphql#792]
## Description of the feature
Greetings! Any way to have ability use `Box<str>` as input type?

## Code example (if possible)
```rust
#[derive(InputObject)]
struct Args {
    username: Box<str>,
    password: Option<Box<str>>,
}
```

**Comment by sunli829** [comment]
I implemented `InputType` for `Box<str>` and `Arc<str>` and released in `v3.0.23`, although I don't know what's the point of you using `Box<str>` instead of `String`. 😁

**Comment by BratSinot** [comment]
> I implemented `InputType` for `Box<str>` and `Arc<str>` and released in `v3.0.23`, 

Thank!

> although I don't know what's the point of you using `Box<str>` instead of `String`. 😁
The answer is simple -- memory. `String` have pointer, lenght and capacity, `str` have only pointer and length. When starting a lot of immutable string data, it saves a lot of memory.
